cranial/connectors/file.py

```diff
@@ -0,0 +1,127 @@
+from datetime import datetime
+import io
+import os
+from tempfile import mkstemp
+from smart_open import open
+import smart_open
+import boto3
+
+# from cranial.connectors import base
+from cranial.common import logger
+
+log = logger.get(name='local_fetchers')  # streaming log
+
+
+def file_readlines(fp, delete_after=False):
+    """
+    memory efficient iterator to read lines from a file (readlines() method reads whole file)
+    Parameters
+    ----------
+    fp
+        path to a decompressed file downloaded from s3 key
+    delete_after
+        delete file after it was read
+    Returns
+    -------
+        generator of lines
+    """
+    for line in open(fp):
+        if line:
+            yield line
+        else:
+            break
+
+    if delete_after:
+        try:
+            os.unlink(fp)
+        except Exception as e:
+            log.warning(e)
+
+
+class Connector(object):
```

Comment:
Should inherit from connectors.base.Connector .

Reply:
connectors.base.Connector is not functional currently because it relies on parsers which have not yet been implemented